### PR TITLE
update all images to present less attackable surface

### PR DESCRIPTION
**Pull request by krancour**
See #1864 

When scanned, alpine-based images produce more favorable results than debian-based images.

Using `gcr.io/distroless/static:nonroot` as a base image, surprisingly, is also regarded as safer than using`scratch` as a base image. (It runs as a non-root user without requiring us to do all the legwork.)

### CODE REVIEW

**Comment by carolynvs** [comment]
What CVE's did it find on scratch? 🤔

**Comment by krancour** [comment]
@carolynvs didn't find any CVEs on scratch, but after researching more, I believe that without additional effort, processes running in a container based on scratch are still running as root. To fix that, it seems the _right_ thing to do is to create a nonroot group and user in the first stage of your build, copy `/etc/passwd` to the final stage, and then use the `USER <whoever>` directive. We could do all of that ourselves, but the `gcr.io/distroless/static:nonroot` does most that legwork for us. (I think we still have to say `USER nonroot:nonroot` in the final stage.)

**Comment by carolynvs** [comment]
Another way to run an image as a nonroot user would be to use `docker run -u1001`. You're right that by default images run as root.root. This is a nicer fix because it doesn't require people to remember to do that though.

**Review by vdice — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -255,19 +277,28 @@ build-logger-windows:
...
+		-t $(DOCKER_IMAGE_PREFIX)$*:$(MUTABLE_DOCKER_TAG) \
 		--build-arg VERSION=$(VERSION) \
 		--build-arg COMMIT=$(GIT_VERSION) \
+		--platform linux/amd64,linux/arm64v8 \
 		.
 
-.PHONY: build-%
-build-%:
-	docker login $(DOCKER_REGISTRY) -u $(DOCKER_USERNAME) -p $${DOCKER_PASSWORD}
+.PHONY: build-alpine-%
```
[the remark this answers is not in the document]
Comment (krancour) [reply Makefile:300]: Good catch.

**Pull request merged**